docs/version_3.md

```diff
@@ -0,0 +1,136 @@
+* https://github.com/amzn/style-dictionary/issues/208
+
+The new build process is similar, except that it does multiple passes of transforming and resolving aliases and defers transforms if the token has an unresolved alias. Each pass might reveal tokens that are ready to be transformed in the next pass. Take this example:
+
+```json
+{
+  "color": {
+    "black": { "value": "#000000" },
+    "font": {
+      "primary": { "value": "{color.black.value}" },
+      "input": { "value": "{color.font.primary.value}" }
+    }
+  }
+}
+```
+
+The first pass will transform the `color.black` token because it doesn't have a reference. It will defer the transforms of `color.font.primary` and `color.font.input` because they do have references. Then it will resolve references to `color.black.value` because it has been transformed.
+
+The second pass will transform `color.font.primary` because it no longer has a reference. It will then resolve the reference to `color.font.primary.value` because it has been transformed. 
+
+The final pass will transform `color.font.input` because it no longer has a reference. Now the build is complete because there are no more deferred transforms left. 
+
+Use cases this change opens up:
+
+* Having variable references in outputs
+* Combining values like using HSL for colors
+* Modifying aliases like making a color lighter or darker
+
+Example (WIP): https://github.com/amzn/style-dictionary/pull/487
+
+Thanks [@mfal](https://github.com/mfal)!
+
+### Custom parser support
+
+https://github.com/amzn/style-dictionary/pull/429
+
+We are pretty excited about this. Until now you could only define your design tokens in either JSON, JSON5, or plain Node modules. The addition of custom parser support allows you to define your tokens in any language you like! The most obvious use-case is to use YAML which has a cleaner and less verbose syntax than JSON. But really the sky is the limit. You can now keep your source of truth in any file format you like as long as you can parse it into an object that Style Dictionary can understand. You register a custom parser like you would a custom transform or format. A parser consists of a pattern to match against files, similar to the test attribute in a loader in Webpack, and a parse function which gets the file path and its contents and is expected to return an object.
+
+* Example: https://github.com/amzn/style-dictionary/tree/3.0/examples/advanced/custom-parser
+* YAML example: https://github.com/amzn/style-dictionary/tree/3.0/examples/advanced/yaml-tokens
+
+TODO: Fix README in example
+
+
+### Adding filePath and isSource entries on tokens
+
+https://github.com/amzn/style-dictionary/pull/356
+
+Style Dictionary adds some metadata on each token before any transforms take place in order to give more context for transforming and formatting tokens. For example, this design token: 
+
+```json
+{
+  "color": {
+    "red": { "value": "#ff0000" }
+  }
+}
+```
+
+Turns into this:
+
+```json
```

Comment:
You can use `json5` as the codeblock syntax to allow the comments to display nicely since `json` doesn't support them.

Using ` ```json `:
```json
{
  "key": "value" // sad comment
}
```
Using ` ```json5 `
```json5
{
  "key": "value" // happy comment
}
```